core/src/main/scala/kafka/tools/ConsumerPerformance.scala

```diff
@@ -302,6 +302,11 @@ object ConsumerPerformance extends LazyLogging {
     val printMetricsOpt = parser.accepts("print-metrics", "Print out the metrics. This only applies to new consumer.")
     val showDetailedStatsOpt = parser.accepts("show-detailed-stats", "If set, stats are reported for each reporting " +
       "interval as configured by reporting-interval")
+    val pollLoopTimeoutOpt = parser.accepts("polling-loop-timeout", "Consumer polling loop timeout.")
```

Comment:
The performance tool is part of the public API, so adding new options requires a short KIP. 

To be honest, the default timeout of 1 second doesn't make much sense to me. I would probably consider either removing the timeout entirely or at least increasing it to something more substantial (say one minute). If we decide to keep it, we should at least add a message to indicate that the tool exited before consuming all messages.

Reply:
Jason, thanks a lot for your remark. I'll be back in a while with new KIP.

Reply:
Please check out KIP-278:

https://cwiki.apache.org/confluence/display/KAFKA/KIP-278%3A+ConsumerPerformance%3A+Increase+Polling+Loop+Timeout+and+Make+It+Reachable+by+the+End+User

Reply:
Thanks! I left a few comments on the dev list.

Reply:
I've updated the KIP number to show the correct number KIP-279
https://cwiki.apache.org/confluence/display/KAFKA/KIP-279%3A+ConsumerPerformance%3A+Increase+Polling+Loop+Timeout+and+Make+It+Reachable+by+the+End+User

Reply:
@hachikuji please, review this [one](http://mail-archives.apache.org/mod_mbox/kafka-dev/201804.mbox/<CAKSnDFST5-%3DvLCrQpPOLjefCyyOBZR-uwe28HrRvrGA9RhJkaA%40mail.gmail.com>) 

@MosheBlumbergX thank you! 

Waiting for some more feedback on KIP-281, friends. 

Thanks

Reply:
@hachikuji No problem, appreciate for your feedback!